### PR TITLE
fix bug zktrie delete

### PR DESCRIPTION
**Pull request by shu-unifra**
```
h1 := t.Hash()
t.TryUpdate(k1,v1)
h2 := t.Hash()
t.TryDelete(k1)
h3 := t.Hash()

```
h1 should equal with h3, but sometimes they did not equal. The reason is that sometimes the trie did not collapsed correctly.
This pr fix this bug.
Test case: https://github.com/scroll-tech/go-ethereum/pull/199

### CODE REVIEW

**Comment by noel2004** [comment]
Awasome work. We have a ['correct' implement for delete in `zktrie_impl.go`](https://github.com/scroll-tech/zktrie/blob/11cc53f2d799d4050baf3a0a3d1b4e4d1f84a606/trie/zk_trie_impl.go#L401) which is not applied yet for waiting more tests. It call the method [`rmAndUpload`](https://github.com/scroll-tech/zktrie/blob/11cc53f2d799d4050baf3a0a3d1b4e4d1f84a606/trie/zk_trie_impl.go#L449). It should be similar to the `delAndUpdate` in this patch.

I would check the two implements and then purge the code for final merging

**Comment by shu-unifra** [comment]
> Awasome work. We have a ['correct' implement for delete in `zktrie_impl.go`](https://github.com/scroll-tech/zktrie/blob/11cc53f2d799d4050baf3a0a3d1b4e4d1f84a606/trie/zk_trie_impl.go#L401) which is not applied yet for waiting for more tests. It call the method [`rmAndUpload`](https://github.com/scroll-tech/zktrie/blob/11cc53f2d799d4050baf3a0a3d1b4e4d1f84a606/trie/zk_trie_impl.go#L449). It should be similar to the `delAndUpdate` in this patch.
> 
> I would check the two implements and then purge the code for final merging

haha, I saw this function, but did not pay too much attention on it. 😂

File: trie/zk_trie_impl.go [hunk 1]
```diff
@@ -124,6 +125,157 @@ func (mt *ZkTrieImpl) TryUpdate(kHash *zkt.Hash, vFlag uint32, vPreimage []zkt.B
 	return nil
 }
 
+func (mt *ZkTrieImpl) delAndUpdate(leaf *Node, path []bool, siblings []*Node) error {
+	leaf.Key()
```
Comment (colinlyguo) [inline trie/zk_trie_impl.go:129]:
```suggestion
```

Comment (colinlyguo) [reply trie/zk_trie_impl.go:129]: Or used for some reasons?

**Comment by noel2004** [comment]
Has update the deletion case in commit [d1c735d4](https://github.com/scroll-tech/zktrie/commit/d1c735d4442a054c64b3ab643479283eae40bb4f). This PR do not pass some additional tests (see https://github.com/scroll-tech/zktrie/tree/unifra/fix_collapsed_bug) so we maily applied the original implement.

It is greatly thanks to @shu-unifra 's contribution to indicate the `sibling type = middle` case which help to fix the original code.

**Pull request closed without merging**